### ISSUE TITLE
support check chunk-name format only in dynamic-import-chunkname rule

### ISSUE THREAD

**Issue by JiangWeixian** [import-js/eslint-plugin-import#2941]
Thanks for awesome works :)

Currently dynamic-import-chunkname rule require webpack chunk name, will report error if not exit.

Is able to add a option to check chunk-name format only, if webpack chunk name not exit is ok.

Currently `dynamic-import-chunkname` will throw error for 

```js
// Bad
import("someModule")

// Good
import(
    /* webpackChunkName: "someModule" */
    'someModule',
  );

// Bad
import(
  /*webpackChunkName:"someModule"*/
  'someModule',
);
```

I want to add a option to make 

```js
// Good
import("someModule")

// Bad
import(
  /*webpackChunkName:"someModule"*/
  'someModule',
);
```

I glad like open a PR for this feature.

Related https://github.com/un-es/eslint-plugin-i/issues/21

**Comment by ljharb** [comment]
You want to *forbid* webpack chunk name comments?

**Comment by JounQin** [comment]
I think he wants to allow empty `leadingComments`:

https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/dynamic-import-chunkname.js#L45-L51

**Comment by ljharb** [comment]
That’s the default behavior of the language without the rule enabled tho.

**Comment by JounQin** [comment]
I think he wants to allow empty `leadingComments` but check the format if it exists?

**Comment by ljharb** [comment]
Hmm, ok - what's the use case? When would you want some dynamic imports handled by webpack but some not?

**Comment by JounQin** [comment]
@ljharb 

Take the following as example, when dynamic string included, a chunk name is good to have.

```js
import(
  /* webpackChunkName: "my-chunk-name" */
  `./locale/${language}`
);
```

**Comment by ljharb** [comment]
The staticness or dynamicness of the import specifier isn't relevant to webpack tho. This rule was created because in a codebase that's using webpack dynamic import code splitting, *every single dynamic import* needs to have a chunk name identified.

**Comment by JiangWeixian** [comment]
@ljharb 

When I use with [loadable](https://loadable-components.com/), e.g.

```ts
loadable(() => import("some-module"))
```

it provides a [babel-plugin](https://loadable-components.com/docs/babel-plugin/) or [swc-plugin](https://www.npmjs.com/package/@swc/plugin-loadable-components) to automatic add `webpackChunkName` during compile time(or swc transform). 

*automatic generate chunkName only enabled when webpackChunkName not found in `leadingComments`* I think it's ok to allow empty `leadingComments`

**Comment by ljharb** [comment]
If you’re using that plugin, why would you use this rule?

**Comment by JiangWeixian** [comment]
If use these plugins:

```
// will automatic add webpackChunkName, /* webpackChunkName: "foo" */
loadable(() => import("foo"))

// will not automatic add webpackChunkName,  /* webpackChunkName: "foo" */
loadable(() => import(/* webpackChunkName: "some-module" */ "some-module"))
```

Because this rule also check webpackChunkName format. If I already defined `webpackChunkName`, I want to check webpackChunkName format is right or not.

**Comment by ljharb** [comment]
OK, thanks, that's at least a use case :-) I'll wait til I hear @JiangWeixian's use case before I decide anything tho.

**Comment by JounQin** [comment]
> OK, thanks, that's at least a use case :-) I'll wait til I hear @JiangWeixian's use case before I decide anything tho.

Not sure what's your meaning, @JiangWeixian has already provide his use case.

**Comment by ljharb** [comment]
I don't see where; the OP only asks for a solution. A use case is the "why" - the problem they're having and why they're having it.

**Comment by JounQin** [comment]
> OK, thanks, that's at least a use case

What does this mean?

**Comment by ljharb** [comment]
It means that you provided the first _motivation_ - reason/justification - for making any changes here, since the OP still hasn't provided any.

**Comment by JounQin** [comment]
<img width="960" alt="image" src="https://github.com/import-js/eslint-plugin-import/assets/8336744/252406bf-7551-47b9-afa9-17faec528644">

I think you meant the OP here? I'm not requesting this feature, just help the OP to explain. I don't have such usage case.

**Comment by ljharb** [comment]
ah thanks, i got confused :-)

In that case, an option that ignores missing comments would be fine.

**Comment by JounQin** [comment]
@JiangWeixian So this feature is ready to be implemented.
